compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -408,10 +408,9 @@ class CheckCaptures extends Recheck, SymTransformer:
       else if meth == defn.Caps_unsafeUnbox then
         mapArgUsing(_.forceBoxStatus(false))
       else if meth == defn.Caps_unsafeBoxFunArg then
-        mapArgUsing:
-          case defn.FunctionOf(paramtpe :: Nil, restpe, isContextual) =>
-            defn.FunctionOf(paramtpe.forceBoxStatus(true) :: Nil, restpe, isContextual)
-
+        mapArgUsing: tp =>
+          val defn.FunctionOf(paramtpe :: Nil, restpe, isContextual) = tp.strippedDealias: @unchecked
```

Comment:
In the previous and current implementations, we need to strip the capture set from `tp` to get to the underlying function type. Is this intended, or are we losing part of the capture set that the resulting function needs?

Reply:
I also noticed that previously we stripped the capture set in `FunctionOf` for `FunctionN` but not for `PolyFunction`. This inconsistency could have been noticed on functions with erased parameters. Now we do not strip capture sets in any of the cases in `FunctionOf`.